src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs

```diff
@@ -165,15 +162,10 @@ protected override void AddEmbeddedResourcesFromAddedModules(ArrayBuilder<Cci.Ma
 
         internal override SynthesizedAttributeData SynthesizeEmbeddedAttribute()
         {
-            if ((object)_lazyEmbeddedAttribute != null)
-            {
```

Comment:
Can we Debug.Assert that _lazyEmbeddedAttribute is not null?

Reply:
Added a comment that the attribute should have been created. I didn't assert since `_lazyEmbeddedAttribute` is dereferenced immediately.

Reply:
IMO, it is easier to just call a see a helpful Debug.Assert message in the exception when something fails. A comment can suffice though.